components/brave_wallet_ui/components/desktop/popup-modals/confirm-password-modal/confirm-password-modal.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+// utils
+import { getLocale } from '../../../../../common/locale'
+
+// hooks
+import { usePasswordAttempts } from '../../../../common/hooks/use-password-attempts'
+
+// components
+import { PopupModal } from '../..'
+import { NavButton } from '../../../extension'
+import { PasswordInput } from '../../../shared'
+
+// style
+import {
+  Row,
+  VerticalSpace
+} from '../../../shared/style'
+import { Title } from '../style'
+import {
+  modalWidth,
+  StyledWrapper
+} from './confirm-password-modal.style'
+
+interface Props {
+  title: string
+  onClose: () => void
+  onSuccess: () => void
+}
+
+export const ConfirmPasswordModal = ({
+  onClose,
+  onSuccess,
+  title
+}: Props) => {
+  // state
+  const [password, setPassword] = React.useState<string>('')
+  const [isCorrectPassword, setIsCorrectPassword] = React.useState<boolean>(true)
+
+  // custom hooks
+  const { attemptPasswordEntry } = usePasswordAttempts({ maxAttempts: 3 })
+
+  // methods
+  const onSubmit = React.useCallback(async () => {
+    if (!password) { // require password to view key
+      return
+    }
+
+    // entered password must be correct
+    const isPasswordValid = await attemptPasswordEntry(password)
+
+    if (!isPasswordValid) {
+      setIsCorrectPassword(isPasswordValid) // set or clear error
+      return // need valid password to continue
+    }
+
+    // clear entered password & error
+    setPassword('')
+    setIsCorrectPassword(true)
+
+    onSuccess()
+  }, [
+    attemptPasswordEntry,
+    password,
+    onSuccess
+  ])
+
+  const onPasswordChange = React.useCallback((value: string): void => {
+    setIsCorrectPassword(true) // clear error
+    setPassword(value)
```

Comment:
if the user somehow locks their keyring right after this (but before they hit enter), should this be cleared?

Reply:
The keyring locked event should redirect the user to the lock screen page, this modal should no longer be visible at this point